### ISSUE TITLE
Is it possible to printer through another device even if one device has printer opened in monitor mode

### ISSUE THREAD

**Issue by mehtahardikr** [star-micronics/StarXpand-SDK-Android#32]
## Description
Device A keep monitoring  the printer. Another Device B wants to print through the same printer. 

## Your device where the bug occurs 
 - Device: 
    *Samsung Galaxy Tab X200*
 - OS:
    *Android 14*

## Your printer
- Model Names:
    *mC-Print3*
- Firmware Version:
    *3.5*
- Interface:
    *LAN*

## Additional context
Is there any way to achieve this ?

**Comment by Tatsuki-Yamamoto2731** [comment]
@mehtahardikr 
When a device monitors a printer, that device occupies the printer's TCP port, preventing other devices from communicating with the printer over TCP. To print from Device B to the printer while Device A is monitoring the printer, you need to stop monitoring on Device A just before printing from Device B. In other words, please close the StarPrinter instance managed by Device A.

**Comment by mehtahardikr** [comment]
For the scanner which is connected to the printer and for its detection we have to keep monitor printer right ?
otherwise we don't have any callback right for inputDevice ?

**Comment by Tatsuki-Yamamoto2731** [comment]
@mehtahardikr 
Let me confirm what you want to achieve. Is the following correct?

- Device A, Device B, and mC-Print3 are on the same network.
- Device A monitors mC-Print3 and detects when the scanner reads something.
- Device B sends print commands to mC-Print3.

If there is only one device, it can handle both printing and detecting scanner readings. However, do you need separate devices for printing and detecting scanner readings?

If YES, the most reasonable method we can propose is as follows:

- Device A does not connect to mC-Print3.
- Device B handles both printing and detecting scanner readings.
- When Device B detects a scanner reading, it sends the obtained data to Device A.

**Comment by Tatsuki-Yamamoto2731** [comment]
@mehtahardikr 
Yes, in order to detect when the scanner reads something, it is necessary to monitor the printer.

Could you please provide more details about the scenario you have in mind?

There are multiple devices that want to connect to the printer. Are the roles of these devices completely separated?
For example:

- Device A monitors the printer and obtains the data read by the scanner. Device A does not send print commands to the printer.
- Device B sends print commands to the printer. Device B does not monitor the printer.

Is your intended scenario like this?
Or do both Device A and Device B handle both monitoring the printer and sending print commands?

**Comment by Tatsuki-Yamamoto2731** [comment]
@mehtahardikr 
Thank you for your information, I understand the senario.

Unfortunately, it is difficult to send print commands from Device B or C while Device A is monitoring the printer. The reason for this is that, as you know, Device A occupies the printer's TCP port, preventing Device B or C from communicating with the printer during that time.

There are two possible solutions to this issue:

Temporarily stop monitoring the printer from Device A only when Device B or C sends print commands. Once the printing from Device B or C is complete, resume monitoring the printer from Device A.
Device B or C does not communicate directly with the printer. When the print button is pressed on Device B or C, Device B or C sends the print data to Device A, and Device A then sends the print data to the printer.

Could you please consider whether it is possible to implement this?

**Comment by Tatsuki-Yamamoto2731** [comment]
@mehtahardikr 
I suggested the solutions for your scenario, and there are no other solutions, unfortunately.
Therefore, there is no further discussion on this issue, and I would like to close this issue. If you are still having problems, please open this issue again.
